src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -664,6 +665,532 @@ public static SyntaxToken XmlEntity(SyntaxTriviaList leading, string text, strin
+        /// Creates the syntax representation of an xml text that contains a newline token with a documentation comment 
+        /// exterior trivia at the end (continued documentation comment).
+        /// </summary>
+        public static XmlTextSyntax XmlNewLine()
+        {
+            return XmlText(XmlTextNewLine());
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml newline token with a documentation comment exterior trivia at 
+        /// the end (continued documentation comment).
+        /// </summary>
+        public static SyntaxToken XmlTextNewLine()
+        {
+            return XmlTextNewLine(true);
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml newline token for xml documentation comments.
+        /// </summary>
+        /// <param name="continueXmlDocumentationComment">
+        /// If set to true, a documentation comment exterior token will be added to the trailing trivia
+        /// of the new token.</param>
+        public static SyntaxToken XmlTextNewLine(bool continueXmlDocumentationComment)
+        {
+            SyntaxToken token = XmlTextNewLine(
+                TriviaList(),
+                Environment.NewLine,
+                Environment.NewLine,
+                TriviaList());
+
+            if (continueXmlDocumentationComment)
+                token = token.WithTrailingTrivia(DocumentationCommentExterior("/// "));
+
+            return token;
+        }
+
+        /// <summary>
+        /// Generates the syntax representation of a xml text node (e.g. for xml documentation comments).
+        /// </summary>
+        /// <param name="value">The string literal used as the text of the xml text node.</param>
+        public static XmlTextSyntax XmlText(string value)
+        {
+            return XmlText(XmlTextLiteral(value));
+        }
+
+        /// <summary>
+        /// Generates the syntax representation of a xml text node (e.g. for xml documentation comments).
+        /// </summary>
+        /// <param name="textTokens">A list of text tokens used as the text of the xml text node.</param>
+        public static XmlTextSyntax XmlText(params SyntaxToken[] textTokens)
+        {
+            return XmlText(TokenList(textTokens));
+        }
+
+        /// <summary>
+        /// Generates the syntax representation of an xml text literal.
+        /// </summary>
+        /// <param name="value">The text used within the xml text literal.</param>
+        public static SyntaxToken XmlTextLiteral(string value)
+        {
+            // TODO: [RobinSedlaczek] It is no compiler hot path here I think. But the contribution guide
+            //       states to avoid LINQ (https://github.com/dotnet/roslyn/wiki/Contributing-Code). With
+            //       XText we have a reference to System.Xml.Linq. Isn't this rule valid here? 
+            string encoded = new XText(value).ToString();
+            return XmlTextLiteral(
```

Comment:
You should be using elastic trivia when creating nodes, so the formatter will correctly format them later.  This is usually automatically specified when calling overloads that do not specify trivia.  It looks like the SyntaxFactory does not have an overload that adds the elastic trivia.  You could add that too.  Look at the implementation of the Literal factory method for an example.